### DISCUSSION TITLE
Annotated dependency that doesn't return a pydantic field type

### DISCUSSION THREAD

**Discussion by gpkc** [fastapi/fastapi#9277] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# Based on https://fastapi.tiangolo.com/tutorial/sql-databases/

from fastapi import APIRouter, Depends
from .database import SessionLocal
from .models import Test as TestModel
from pydantic import BaseModel

router = APIRouter(prefix="/test", tags=["test"])

def get_db():
    db = SessionLocal()
    try:
        yield db
    finally:
        db.close()

DatabaseDependency = Annotated[Session, Depends(get_db)]

class TestSchema(BaseModel):
    test: str | None = None

    class Config:
        orm_mode = True

# Does not work
@router.get("/get-tests", response_model=List[TestSchema])
async def get_tests(db: DatabaseDependency):
    tests = TestModel.get_all()
    return tests

# Works
@router.get("/get-tests", response_model=List[TestSchema])
async def get_tests(db: Session = Depends(get_db)):
    tests = TestModel.get_all()
    return tests
```

### Description

This raises `fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that typing.Annotated[sqlalchemy.orm.session.Session, Depends(get_db)] is a valid Pydantic field type. If you are using a return type annotation that is not a valid Pydantic field (e.g. Union[Response, dict, None]) you can disable generating the response model from the type annotation with the path operation decorator parameter response_model=None. Read more: https://fastapi.tiangolo.com/tutorial/response-model/`

It is unclear from the docs for Annotated and response_model why using an annotated dependency would influence the response model specifically only when using Annotated.

### Operating System

Linux

### FastAPI Version

0.94.1

### Python Version

Python 3.11.2

**Comment by masyagin1998** [comment]
Issuing this problem too (((

**Comment by Viicos** [comment, marked as answer]
Upgrade to FastAPI 0.95.0.

**Comment by gpkc** [comment]


**Reply by Viicos** [reply]
https://github.com/tiangolo/fastapi/pull/4871 was introduced in 0.95, but looking at the last comments on this PR you're not the only one that forgot to upgrade 😄 (as all docs examples have been updated to use `Annotated`, this was excepted).

**Reply by tiangolo** [reply]
Thanks for the help @Viicos! There's now a note in the docs sections for `Annotated` to explain that it is available since FastAPI 0.95.0 :nerd_face: 

https://github.com/tiangolo/fastapi/pull/9298
